### ISSUE TITLE
Test failure: org.eclipse.jetty.osgi.test.TestJettyOSGiBootHTTP2.testHTTP2

### ISSUE THREAD

**Issue by joakime** [eclipse/jetty.project#2247]
`jetty-9.4.x` branch.
Linux build.
Java 8 build.
Bare metal build. (no docker / container)

```
java.lang.AssertionError
	at org.eclipse.jetty.osgi.test.TestJettyOSGiBootHTTP2.testHTTP2(TestJettyOSGiBootHTTP2.java:156)
```

**Comment by joakime** [comment]
Full log for `test-jetty-osgi` testing ...

```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.eclipse.jetty.osgi.test.TestJettyOSGiBootWithAnnotations

org.eclipse.jetty.util[org.eclipse.jetty.util.log] : test-spec webapp is deployed. DO NOT USE IN PRODUCTION! Ignored FQCN: org.eclipse.jetty.util.log.Slf4jLog

[WARNING] Tests run: 2, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 3.409 s - in org.eclipse.jetty.osgi.test.TestJettyOSGiBootWithAnnotations
[INFO] Running org.eclipse.jetty.osgi.test.TestJettyOSGiBootWebAppAsService


[WARNING] Tests run: 2, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 2.319 s - in org.eclipse.jetty.osgi.test.TestJettyOSGiBootWebAppAsService
[INFO] Running org.eclipse.jetty.osgi.test.TestJettyOSGiBootWithWebSocket

org.eclipse.jetty.util[org.eclipse.jetty.util.DeprecationWarning] : Using @Deprecated Class org.eclipse.jetty.servlets.MultiPartFilter Ignored FQCN: org.eclipse.jetty.util.log.Slf4jLog

[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.283 s - in org.eclipse.jetty.osgi.test.TestJettyOSGiBootWithWebSocket
[INFO] Running org.eclipse.jetty.osgi.test.TestJettyOSGiBootWithJsp

org.eclipse.jetty.util[org.eclipse.jetty.util.DeprecationWarning] : Using @Deprecated Class org.eclipse.jetty.servlets.MultiPartFilter Ignored FQCN: org.eclipse.jetty.util.log.Slf4jLog

[WARNING] Tests run: 2, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 2.694 s - in org.eclipse.jetty.osgi.test.TestJettyOSGiBootWithJsp
[INFO] Running org.eclipse.jetty.osgi.test.TestJettyOSGiBootWithJavaxWebSocket

org.eclipse.jetty.util[org.eclipse.jetty.util.DeprecationWarning] : Using @Deprecated Class org.eclipse.jetty.servlets.MultiPartFilter Ignored FQCN: org.eclipse.jetty.util.log.Slf4jLog

[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 3.389 s - in org.eclipse.jetty.osgi.test.TestJettyOSGiBootWithJavaxWebSocket
[INFO] Running org.eclipse.jetty.osgi.test.TestJettyOSGiBootHTTP2Conscrypt

org.eclipse.jetty.util[org.eclipse.jetty.util.DeprecationWarning] : Using @Deprecated Class org.eclipse.jetty.servlets.MultiPartFilter Ignored FQCN: org.eclipse.jetty.util.log.Slf4jLog

[WARNING] Tests run: 2, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 3.72 s - in org.eclipse.jetty.osgi.test.TestJettyOSGiBootHTTP2Conscrypt
[INFO] Running org.eclipse.jetty.osgi.test.TestJettyOSGiBootContextAsService


[WARNING] Tests run: 2, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 1.534 s - in org.eclipse.jetty.osgi.test.TestJettyOSGiBootContextAsService
[INFO] Running org.eclipse.jetty.osgi.test.TestJettyOSGiBootHTTP2

org.eclipse.jetty.util[org.eclipse.jetty.osgi.boot.internal.serverfactory.JettyServerServiceTracker] :  Ignored FQCN: org.eclipse.jetty.util.log.Slf4jLog
java.net.BindException: Address already in use
	at sun.nio.ch.Net.bind0(Native Method)
	at sun.nio.ch.Net.bind(Net.java:433)
	at sun.nio.ch.Net.bind(Net.java:425)
	at sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:223)
	at sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:74)
	at org.eclipse.jetty.server.ServerConnector.openAcceptChannel(ServerConnector.java:340)
	at org.eclipse.jetty.server.ServerConnector.open(ServerConnector.java:308)
	at org.eclipse.jetty.server.AbstractNetworkConnector.doStart(AbstractNetworkConnector.java:80)
	at org.eclipse.jetty.server.ServerConnector.doStart(ServerConnector.java:244)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
	at org.eclipse.jetty.server.Server.doStart(Server.java:398)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
	at org.eclipse.jetty.osgi.boot.internal.serverfactory.ServerInstanceWrapper.start(ServerInstanceWrapper.java:296)
	at org.eclipse.jetty.osgi.boot.internal.serverfactory.JettyServerServiceTracker.addingService(JettyServerServiceTracker.java:63)
	at org.osgi.util.tracker.ServiceTracker$Tracked.customizerAdding(ServiceTracker.java:941)
	at org.osgi.util.tracker.ServiceTracker$Tracked.customizerAdding(ServiceTracker.java:1)
	at org.osgi.util.tracker.AbstractTracked.trackAdding(AbstractTracked.java:256)
	at org.osgi.util.tracker.AbstractTracked.track(AbstractTracked.java:229)
	at org.osgi.util.tracker.ServiceTracker$Tracked.serviceChanged(ServiceTracker.java:901)
	at org.eclipse.osgi.internal.serviceregistry.FilteredServiceListener.serviceChanged(FilteredServiceListener.java:109)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.dispatchEvent(BundleContextImpl.java:920)
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)
	at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:148)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEventPrivileged(ServiceRegistry.java:862)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEvent(ServiceRegistry.java:801)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.register(ServiceRegistrationImpl.java:127)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.registerService(ServiceRegistry.java:225)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.registerService(BundleContextImpl.java:469)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.registerService(BundleContextImpl.java:487)
	at org.eclipse.jetty.osgi.boot.internal.serverfactory.DefaultJettyAtJettyHomeHelper.startJettyAtJettyHome(DefaultJettyAtJettyHomeHelper.java:184)
	at org.eclipse.jetty.osgi.boot.JettyBootstrapActivator.start(JettyBootstrapActivator.java:82)
	at org.eclipse.osgi.internal.framework.BundleContextImpl$3.run(BundleContextImpl.java:779)
	at org.eclipse.osgi.internal.framework.BundleContextImpl$3.run(BundleContextImpl.java:1)
	at java.security.AccessController.doPrivileged(Native Method)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.startActivator(BundleContextImpl.java:772)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.start(BundleContextImpl.java:729)
	at org.eclipse.osgi.internal.framework.EquinoxBundle.startWorker0(EquinoxBundle.java:933)
	at org.eclipse.osgi.internal.framework.EquinoxBundle$EquinoxModule.startWorker(EquinoxBundle.java:309)
	at org.eclipse.osgi.container.Module.doStart(Module.java:581)
	at org.eclipse.osgi.container.Module.start(Module.java:449)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.incStartLevel(ModuleContainer.java:1634)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.incStartLevel(ModuleContainer.java:1614)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.doContainerStartLevel(ModuleContainer.java:1585)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.dispatchEvent(ModuleContainer.java:1528)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.dispatchEvent(ModuleContainer.java:1)
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)
	at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:340)

[ERROR] Tests run: 3, Failures: 1, Errors: 0, Skipped: 2, Time elapsed: 1.54 s <<< FAILURE! - in org.eclipse.jetty.osgi.test.TestJettyOSGiBootHTTP2
[ERROR] testHTTP2(org.eclipse.jetty.osgi.test.TestJettyOSGiBootHTTP2)  Time elapsed: 0.066 s  <<< FAILURE!
java.lang.AssertionError
	at org.eclipse.jetty.osgi.test.TestJettyOSGiBootHTTP2.testHTTP2(TestJettyOSGiBootHTTP2.java:156)
```

There does not appear to be any dead / zombie processes on the `linux-nuc-1` device that can account for the above behavior.

**Comment by joakime** [comment]
The assertion is from ...

https://github.com/eclipse/jetty.project/blob/be64f322add1f86404f8c2f15c5e67b7e5de7c5b/jetty-osgi/test-jetty-osgi/src/test/java/org/eclipse/jetty/osgi/test/TestJettyOSGiBootHTTP2.java#L154-L156

**Comment by janbartel** [comment]
Fixed in jetty-9.4 and above.
